config/config_encryption_test.go

```diff
@@ -106,8 +118,182 @@ func TestRemoveECS(t *testing.T) {
+	defer func() { os.Stdin = oldIn }()
+	os.Stdin, err = os.Open(passFile.Name())
+	if err != nil {
+		t.Fatalf("Problem opening temp file at %s: %s\n", passFile.Name(), err)
+	}
+
+	// Save encrypted config
+	enc1 := filepath.Join(tempDir, "encrypted.dat")
+	err = c.SaveConfig(enc1, false)
+	if err != nil {
+		t.Fatalf("Problem storing config in file %s: %s\n", enc1, err)
+	}
+	// Save again
+	enc2 := filepath.Join(tempDir, "encrypted2.dat")
+	err = c.SaveConfig(enc2, false)
+	if err != nil {
+		t.Fatalf("Problem storing config in file %s: %s\n", enc2, err)
+	}
+	data1, err := ioutil.ReadFile(enc1)
+	if err != nil {
+		t.Fatalf("Problem reading file %s: %s\n", enc1, err)
+	}
+	data2, err := ioutil.ReadFile(enc2)
+	if err != nil {
+		t.Fatalf("Problem reading file %s: %s\n", enc2, err)
+	}
+	// legth of prefix + salt
+	l := len(EncryptConfirmString+SaltPrefix) + SaltRandomLength
+	// Even though prefix, including salt with the random bytes is the same
+	if !bytes.Equal(data1[:l], data2[:l]) {
+		t.Error("Salt is not reused.")
+	}
+	// the cipher text should not be
+	if bytes.Equal(data1, data2) {
+		t.Error("Encryption key must have been reused as cipher texts are the same")
+	}
+}
+
+func TestSaveAndReopenEncryptedConfig(t *testing.T) {
+	c := &Config{}
+	c.Name = "myCustomName"
+	c.EncryptConfig = 1
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("Problem creating temp dir at %s: %s\n", tempDir, err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Prepare password
+	passFile, err := ioutil.TempFile(tempDir, "*.pw")
+	if err != nil {
+		t.Fatalf("Problem creating temp file at %s: %s\n", tempDir, err)
+	}
+	passFile.WriteString("pass\npass\n")
+	passFile.Close()
+
+	// Temporarily replace Stdin with a custom input
+	cleanup := setAnswersFile(t, passFile.Name())
+	defer cleanup()
+
+	// Save encrypted config
+	enc := filepath.Join(tempDir, "encrypted.dat")
+	err = c.SaveConfig(enc, false)
+	if err != nil {
+		t.Fatalf("Problem storing config in file %s: %s\n", enc, err)
+	}
+
+	// Prepare password input for decryption
+	passFile, err = os.Open(passFile.Name())
+	if err != nil {
+		t.Fatalf("Problem opening temp file at %s: %s\n", passFile.Name(), err)
+	}
+	defer passFile.Close()
+	os.Stdin = passFile
+
+	// Clean session
+	readConf := &Config{}
+	// Load with no existing state, key is read from the prepared file
+	err = readConf.ReadConfig(enc, true)
+
+	// Verify
+	if err != nil {
+		t.Fatalf("Problem reading config in file %s: %s\n", enc, err)
+	}
+
+	if c.Name != readConf.Name || c.EncryptConfig != readConf.EncryptConfig {
+		t.Error("Loaded conf not the same as original")
+	}
+}
+
+// setAnswersFile sets the given file as the current stdin
+// returns the close function to defer for reverting the stdin
+func setAnswersFile(t *testing.T, answerFile string) func() {
+	oldIn := os.Stdin
+
+	inputFile, err := os.Open(answerFile)
+	if err != nil {
+		t.Fatalf("Problem opening temp file at %s: %s\n", inputFile.Name(), err)
```

Comment:
There is a chance that inputFile might be nil here should this be answerFile instead?

Reply:
Done.